internals/overlord/overlord.go

```diff
@@ -379,15 +371,34 @@ func (o *Overlord) Loop() {
 			}
 		}
 	})
+	o.ensureWaitRun()
```

Comment:
I am also discussing this proposal with @pedronis in a separate thread, to make sure this is compatible with snapd.